game/README.md

```diff
@@ -6,7 +6,263 @@
+
+```yml
+services:
+  "gameProvider.basketballService":
+    registry: "demoZk"
+    protocol: "dubbo"
+    interface: "org.apache.dubbo.game.BasketballService"
+    loadbalance: "random"
+    warmup: "100"
+    cluster: "failover"
+    methods:
+      - name: "Online"
+        retries: 0
+      - name: "Offline"
+        retries: 0
+      - name: "Message"
+        retries: 
+```
+
+#### consumer side
+
+The consumer of basketball is mainly used to be called by gate, so the main code is put in game/pkg/consumer/gate/basketball.go,  The consumer code is as follows.
+
+```go
+type BasketballService struct {
+    Send func(ctx context.Context, uid string, data string) (*pojo.Result, error)
+}
+
+func (p *BasketballService) Reference() string {
+    return "gateConsumer.basketballService"
+}
+```
+
+In basketball, you only need to instantiate a consumer variable.
+
+```go
+var gateBasketball = new(gate.BasketballService)
+```
+
+Then register to Dubbo go in main.
+
+```go
+config.SetConsumerService(gateBasketball)
+```
+
+##### configuration file
+
+```yml
+references:
+  "gateConsumer.basketballService":
+    registry: "demoZk"
+    protocol : "dubbo"
+    interface : "org.apache.dubbo.gate.BasketballService"
+    cluster: "failover"
+    methods:
+      - name: "Send"
+        retries: 0
+```
+
+Because the consumer of game needs to call the provider of gate, the string returned by the reference method is gateConsumer.basketballService  , in the configuration file gateConsumer.basketballService  And game/pkg/consumer/gate/basketball.go  The value of intreface should be consistent with that of the provider of gate.
+
+### gate module
+
+#### server side
+
+```go
+type BasketballService struct{}
+
+func (p *BasketballService) Send(ctx context.Context, uid, data string) (*pojo.Result, error) {
+...
+}
+
+func (p *BasketballService) Reference() string {
+    return "gateProvider.basketballService"
+}
+```
+
+Register with Dubbo
+
+```go
+config.SetProviderService(new(BasketballService))
+```
+
+##### configuration file
+
+gateProvider.basketballService  The interface here must be the same as that in game client.yml  The settings in the file should be consistent, otherwise game cannot send data to gate.
+
+```yml
+# service config
+services:
+  "gateProvider.basketballService":
+    registry: "demoZk"
+    protocol : "dubbo"
+    interface : "org.apache.dubbo.gate.BasketballService"
+    loadbalance: "random"
+    warmup: "100"
+    cluster: "failover"
+    methods:
+      - name: "Send"
+        retries: 0
+```
+
+#### consumer size
+
+The consumer side of the gate is relatively special. Because the consumer of the gate needs to call the service in the game, in the game, the consumer directly instantiates the service of a game, and its method directly uses the instantiated object gamebasketball to call, thus realizing the function of a gateway.
```

Comment:
The consumer side of the gate is relatively special. Because the consumer of the gate needs to call the service in the game,  the consumer directly instantiates the service of a game, and its method directly uses the instantiated object gamebasketball to call, thus realizing the function of a gateway.